File: code/degrees_separation.R
# Encode degrees of separation
## Initialize objects
cite_degrees <- list()
i <- 1
## Create vectors for get_works
while(cited_by > 0 || cites > 0){
  # Check for 0 or more degrees
  cited_by_i <- min(cited_by, 1)
  cites_i <- min(cites, 1)
  # Create the ith vector
  cite_degrees[[i]] <- c(rep('cited_by', cited_by_i),
                         rep('cites', cites_i))
  # Increment/Decrement counts
  i <- i + 1
  cited_by <- cited_by - cited_by_i
  cites <- cites - cites_i
}

# Clean up temporary objects
rm('cited_by', 'cited_by_i', 'cites', 'cites_i', 'i')
gc()

File: code/degrees_separation.py
# -*- coding: utf-8 -*-
"""
Created on Wed Mar  5 11:11:53 2025

@author: kevinatpenn
"""

import gc

# Encode degrees of separation
## Initialize objects
cite_degrees = []
deg = 1

## Create vectors for get_works
while cited_by > 0 or cites > 0:
    # Check for 0 or more degrees
    cited_by_deg = min(cited_by, 1)
    cites_deg = min(cites, 1)
    # Create a vector for the current degree of separation
    cite_degrees.append(['cited_by'] * cited_by_deg + ['cites'] * cites_deg)
    # Increment/Decrement counts
    deg += 1
    cited_by -= cited_by_deg
    cites -= cites_deg

# Clean up temporary objects
del cited_by, cited_by_deg, cites, cites_deg, deg
gc.collect()
